contrib/go/src/python/pants/contrib/go/tasks/go_compile.py

```diff
@@ -82,9 +98,10 @@ def _sync_binary_dep_links(self, target, gopath, lib_binary_map):
                                              library.
 
     Required links to binary dependencies under gopath's "pkg/" dir are either created if
-    non-existent, or refreshed if the link is older than the underlying binary. Any pre-existing
-    links within gopath's "pkg/" dir that do not correspond to a transitive dependency of target
-    are deleted.
+    non-existent, refreshed if the link is older than the underlying binary, or replaced with a
```

Comment:
This smells like the invalidation bug mentioned above and likely can be reverted when that is fixed.

Reply:
Confirmed this is fixed after updating the fingerprint strategy, so backed this out.